tests/unit/assessment/test_dashboards.py

```diff
@@ -0,0 +1,647 @@
+
+    crawler.snapshot()
+
+    rows = mock_backend.rows_written_for("hive_metastore.test.redash_dashboards", "overwrite")
+    assert rows == [Row(id="did1", name=None, parent=None, query_ids=[], tags=[], creator_id=None)]
+    ws.dashboards.list.assert_called_once()
+
+
+def test_redash_dashboard_crawler_includes_dashboard_ids(mock_backend) -> None:
+    ws = create_autospec(WorkspaceClient)
+    ws.dashboards.get.return_value = SdkRedashDashboard(id="did1")
+    crawler = RedashDashboardCrawler(ws, mock_backend, "test", include_dashboard_ids=["did1"])
+
+    crawler.snapshot()
+
+    rows = mock_backend.rows_written_for("hive_metastore.test.redash_dashboards", "overwrite")
+    assert rows == [Row(id="did1", name=None, parent=None, query_ids=[], tags=[], creator_id=None)]
+    ws.dashboards.get.assert_called_once_with("did1")
+    ws.dashboards.list.assert_not_called()
+
+
+def test_redash_dashboard_crawler_skips_not_found_dashboard_ids(caplog, mock_backend) -> None:
+    ws = create_autospec(WorkspaceClient)
+
+    def get_dashboards(dashboard_id: str) -> SdkRedashDashboard:
+        if dashboard_id == "did1":
+            return SdkRedashDashboard(id="did1")
+        raise NotFound(f"Did not find dashboard: {dashboard_id}")
+
+    ws.dashboards.get.side_effect = get_dashboards
+    crawler = RedashDashboardCrawler(ws, mock_backend, "test", include_dashboard_ids=["did1", "did2"])
+
+    with caplog.at_level(logging.WARNING, logger="databricks.labs.ucx.assessment.dashboards"):
+        crawler.snapshot()
+
+    rows = mock_backend.rows_written_for("hive_metastore.test.redash_dashboards", "overwrite")
+    assert rows == [Row(id="did1", name=None, parent=None, query_ids=[], tags=[], creator_id=None)]
+    assert "Cannot get Redash dashboard: did2" in caplog.messages
+    ws.dashboards.get.assert_has_calls([call("did1"), call("did2")])
+    ws.dashboards.list.assert_not_called()
+
+
+def list_legacy_queries() -> list[LegacyQuery]:
+    queries = [
+        LegacyQuery(id="qid1", name="First query", parent="parent", query="SELECT 42 AS count"),
+        LegacyQuery(id="qid2", name="Second query", parent="parent", query="SELECT 21 AS count"),
+    ]
+    return queries
+
+
+def get_legacy_query(query_id: str) -> LegacyQuery:
+    for query in list_legacy_queries():
+        if query.id == query_id:
+            return query
+    raise NotFound(f"Legacy query: {query_id}")
+
+
+def test_redash_dashboard_crawler_list_queries_includes_query_ids(mock_backend) -> None:
+    ws = create_autospec(WorkspaceClient)
+    ws.queries_legacy.list.side_effect = list_legacy_queries
+    ws.queries_legacy.get.side_effect = get_legacy_query
+    crawler = RedashDashboardCrawler(ws, mock_backend, "test", include_query_ids=["qid1"])
+
+    queries = list(crawler.list_queries())
+
+    assert queries == [Query(id="qid1", name="First query", parent="parent", query="SELECT 42 AS count")]
```

Comment:
😍

Reply:
Yes,  me being inconsistent :/ Thanks for reviewing!